src/ordered_robin_dict.jl

```diff
@@ -0,0 +1,476 @@
+const ALLOWABLE_USELESS_GROWTH = 0.25
+
+"""
+    OrderedRobinDict([itr])
+
+`OrderedRobinDict{K,V}()` constructs a ordered dictionary with keys of type `K` and values of type `V`.
+It takes advantage of `RobinDict` in maintaining the order of the keys. 
+Given a single iterable argument, constructs a [`OrderedRobinDict`](@ref) whose key-value pairs 
+are taken from 2-tuples `(key,value)` generated by the argument.
+
+
+# Examples
+```jldoctest
+julia> OrderedRobinDict([("A", 1), ("B", 2)])
+OrderedRobinDict{String,Int64} with 2 entries:
+  "A" => 1
+  "B" => 2
+```
+
+Alternatively, a sequence of pair arguments may be passed.
+
+```jldoctest
+julia> OrderedRobinDict("A"=>1, "B"=>2)
+OrderedRobinDict{String,Int64} with 2 entries:
+  "A" => 1
+  "B" => 2
+```
+"""
+mutable struct OrderedRobinDict{K,V} <: AbstractDict{K,V}
+    dict::RobinDict{K, Int32} 
+    keys::Vector{K}
+    vals::Vector{V}
+    count::Int32
+
+    function OrderedRobinDict{K, V}() where {K, V}
+        new{K, V}(RobinDict{K, Int32}(), Vector{K}(), Vector{V}(), 0)
+    end
+
+    function OrderedRobinDict{K, V}(d::OrderedRobinDict{K, V}) where {K, V}
+        new{K, V}(copy(d.dict), copy(d.keys), copy(d.vals), d.count)
+    end
+
+    function OrderedRobinDict{K,V}(kv) where {K, V}
+        h = OrderedRobinDict{K,V}()
```

Comment:
Couild we be using iterator traits to check if this has length or size, and if so do `sizehint!(h, length(kv))` ?